youtube_dl/extractor/common.py

```diff
@@ -2167,7 +2167,14 @@ def _find_jwplayer_data(webpage):
             r'jwplayer\((?P<quote>[\'"])[^\'" ]+(?P=quote)\)\.setup\s*\((?P<options>[^)]+)\)',
             webpage)
         if mobj:
-            return mobj.group('options')
+            options = mobj.group('options').strip()
+            # We're looking for JSON
+            if options[0] != '{':
+                # setup() got a variable as an argument - resolve it
+                regex = r'%s\s*=\s*(?P<options>\{[^\}]\})' % options
```

Comment:
1. ```options``` should be escaped - ```re.escape```
2. Nested variables won't be detected. For example: (from https://developer.jwplayer.com/jw-player/docs/developer-guide/customization/configuration-reference/)
```
var config = {
    "playlist": [{
        "file": "/assets/sintel.mp4",
        "image": "/assets/sintel.jpg",
        "title": "Sintel Trailer",
        "mediaid": "ddra573"
    },{
        "file": "/assets/bigbuckbunny.mp4",
        "image": "/assets/bigbuckbunny.jpg",
        "title": "Big Buck Bunny Trailer",
        "mediaid": "ddrx3v2"
    }]
};
```